frontend/src/js/components/editors/template/TemplateLabelSettings.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+
+
+var TemplateLabelSettings = React.createClass({
+    propTypes: {
+        label: React.PropTypes.object,
+        update: React.PropTypes.func
+    },
+    updateLabelValue(key) {
+        const self = this;
+        let validator = parseFloat;
+        if (key === "priority") {
+            validator = parseInt;
+        }
+        return (e) => {
+            self.props.update(key, validator(this.range(e.target.value)));
+        };
+    },
+    range(alt){
+        if(alt < 0){
+            alt = 0;
+        }
+        else if (alt > 1){
+            alt = 1;
+        }
+        return alt;
+    },
```

Comment:
Yeah, the general approach is "right", but we need to change the specifics. Basically, you want to use the onChange handler to "validate" the data and make sure it's good (integer for priority,  0<= float <=1 for others); you store the string value in this.state (using `this.setState({"key": "value"})`), and have a separate state value of "valid" that's a boolean; if a field is invalid, you stick a class on the field that will mark it as so. Then you can use `onBlur` to actually sync the value out to the parent.